contribs/gnofaucet/github.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"math"
+	"os"
+	"time"
+
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/redis/go-redis/v9"
+)
+
+type githubCfg struct {
+	rootCfg           *serveCfg
+	ghClientID        string
+	maxClaimableLimit int64
+	cooldownPeriod    time.Duration
+}
+
+func (c *githubCfg) RegisterFlags(fs *flag.FlagSet) {
+	fs.StringVar(
+		&c.ghClientID,
+		"github-client-id",
+		"",
+		"github client id for oauth authentication",
+	)
+
+	fs.DurationVar(
+		&c.cooldownPeriod,
+		"cooldown-period",
+		24*time.Hour,
+		"minimum required time between consecutive faucet claims by the same user",
+	)
+
+	fs.Int64Var(
+		&c.maxClaimableLimit,
+		"max-claimable-limit",
+		math.MaxInt64,
+		"maximum number of tokens a single user can claim over their lifetime",
+	)
+}
+
+func newGithubCmd(rootCfg *serveCfg) *commands.Command {
+	cfg := &githubCfg{
+		rootCfg: rootCfg,
+	}
+
+	return commands.NewCommand(
+		commands.Metadata{
+			Name:       "github",
+			ShortUsage: "github [flags]",
+			LongHelp:   "applies github middleware to the gno.land faucet",
+		},
+		cfg,
+		func(ctx context.Context, args []string) error {
+			return execGithub(ctx, cfg, commands.NewDefaultIO())
+		},
+	)
+}
+
+func execGithub(ctx context.Context, cfg *githubCfg, io commands.IO) error {
+	if cfg.ghClientID == "" {
+		return fmt.Errorf("github client id is required")
```

Comment:
These should be standard errors

Reply:
Changed to standard errors 82c3600f317e8a29df77f24290a24298d7332314